src/transformStateWithClones.js

```diff
@@ -7,7 +7,38 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const result = [];
+  let newObject = Object.assign({}, state);
+
+  for (const object of actions) {
+    switch (object.type) {
+      case 'addProperties':
+        newObject = Object.assign(newObject, object.extraData);
+        result.push(Object.assign({}, newObject));
```

Comment:
you do the same thing (pushing result object to some array) in each case

try to move this logic outside of the case functionality (do this by default after all cases)